src/AutoMapper/IMappingEngine.cs

```diff
@@ -172,11 +160,20 @@ public interface IMappingEngine
         /// <param name="destination"></param>
         /// <param name="sourceType">Source type to use</param>
         /// <param name="destinationType">Destination type to use</param>
-        [Obsolete("Set the CreateMissingTypeMaps property on Mapper.ConfigurationProvider or your Profile instead")]
         void DynamicMap(object source, object destination, Type sourceType, Type destinationType);
 
+        /// <summary>
+        /// Configuration provider for performaing maps
+        /// </summary>
+        IConfigurationProvider ConfigurationProvider { get; }
```

Comment:
It is, the IConfigurationProvider interface is now complete segregated from the mutable side.